### PR TITLE
Fix building library with Ktor 2

### PR DESCRIPTION
**Pull request by cmelchior**
After moving to Ktor2, using Realm in real projects didn't work. This was caught by our integration tests added after moving to the new project structure.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -198,7 +198,7 @@ pipeline {
                             // This will overwrite previous test results, but should be ok as we would not get here
                             // if previous stages failed.
                             {
-                                testAndCollect("packages", "cleanAllTests macosTest -PincludeSdkModules=false -Pkotlin.native.binary.memoryModel=experimental")
+                                testAndCollect("packages", "cleanAllTests macosTest -PincludeSdkModules=false -Pkotlin.native.binary.freezing=disabled -Pkotlin.native.binary.memoryModel=experimental")
```
Comment (nhachicha) [inline Jenkinsfile:201]: This shouldn't be needed once we merge https://github.com/realm/realm-kotlin/pull/1059

File: Jenkinsfile [hunk 2]
```diff
@@ -218,7 +218,7 @@ pipeline {
                     steps {
                         testWithServer([
                             {
-                                testAndCollect("packages", "cleanAllTests -PincludeSdkModules=false iosTest")
+                                testAndCollect("packages", "cleanAllTests iosTest -PincludeSdkModules=false -Pkotlin.native.binary.freezing=disabled -Pkotlin.native.binary.memoryModel=experimental")
```
Comment (nhachicha) [inline Jenkinsfile:221]: ditto

**Review by nhachicha — APPROVED** [review]
LGTM 👍

File: Jenkinsfile (hunk 2 above)
Comment (cmelchior) [reply Jenkinsfile:221]: See https://github.com/realm/realm-kotlin/pull/1043#discussion_r991332626

**Pull request merged**